src/wallet/rpcwallet.cpp

```diff
@@ -2359,3 +2361,293 @@ UniValue listunspent(const UniValue& params, bool fHelp)
+example "m/44'/0'/0'/c" will result in m/44'/0'/0'/0/1 for the second external key
+example "m/44'/0'/0'/c" will result in m/44'/0'/0'/1/1 for the second internal key
+*/
+const std::string hd_default_chainpath = "m/44'/0'/0'/c";
+
+static void SendMoneyHD(const CTxDestination &address, CAmount nValue, bool fSubtractFeeFromAmount, CWalletTx& wtxNew)
+{
+    CAmount curBalance = pwalletMain->GetBalance();
+
+    // Check amount
+    if (nValue <= 0)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid amount");
+
+    if (nValue > curBalance)
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Insufficient funds");
+
+    // Parse Bitcoin address
+    CScript scriptPubKey = GetScriptForDestination(address);
+
+    // Create and send the transaction
+    CHDReserveKey reservekey(pwalletMain);
+    CAmount nFeeRequired;
+    std::string strError;
+    vector<CRecipient> vecSend;
+    int nChangePosRet = -1;
+    CRecipient recipient = {scriptPubKey, nValue, fSubtractFeeFromAmount};
+    vecSend.push_back(recipient);
+    if (!pwalletMain->CreateTransaction(vecSend, wtxNew, reservekey, nFeeRequired, nChangePosRet, strError)) {
+        if (!fSubtractFeeFromAmount && nValue + nFeeRequired > pwalletMain->GetBalance())
+            strError = strprintf("Error: This transaction requires a transaction fee of at least %s because of its amount, complexity, or use of recently received funds!", FormatMoney(nFeeRequired));
+        throw JSONRPCError(RPC_WALLET_ERROR, strError);
+    }
+    if (!pwalletMain->CommitTransaction(wtxNew, reservekey))
+        throw JSONRPCError(RPC_WALLET_ERROR, "Error: The transaction was rejected! This might happen if some of the coins in your wallet were already spent, such as if you used a copy of wallet.dat and coins were spent in the copy but not marked as spent here.");
+}
+
+UniValue hdaddchain(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp)
+        throw runtime_error(
+                            "hdaddchain (<chainpath>|default) (<masterseed_hex>)\n"
+                            "\nAdds a HD/Bip32 chain \n"
+                            "\nArguments:\n"
+                            "1. chainpath        (string, optional, default="+hd_default_chainpath+") chainpath for hd wallet structure\n"
+                            "   m stands for master, c for internal/external key-switch, k stands for upcounting child key index"
+                            "2. masterseed_hex   (string/hex, optional) use this seed for master key generation\n"
+                            "\nResult\n"
+                            "{\n"
+                            "  \"seed_hex\" : \"<hexstr>\",  (string) seed used during master key generation (only if no masterseed hex was provided\n"
+                            "}\n"
+
+                            "\nExamples\n"
+                            + HelpExampleCli("hdaddchain", "set")
```

Comment:
should this not read `HelpExampleCli("hdaddchain", "")` ? Since by providing "set" as an argument, you must also provide a seed ... I believe the intention was to provide a example run without providing a seed, which should just run as `hdaddchain` without arguments